model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -213,6 +214,26 @@ class ModelReplicationServer(
             val newVersionHash = repositoriesManager.pollVersionHash(branchRef(), lastKnownVersionHash)
             call.respondDelta(newVersionHash, lastKnownVersionHash)
         }
+        post<Paths.postRepositoryObjectsGetAll> {
+            val repository = RepositoryId(call.parameters["repository"]!!)
+            val keys = call.receiveStream().bufferedReader().use { reader ->
+                reader.lineSequence().toHashSet()
+            }
+            val objects = withContext(Dispatchers.IO) { modelClient.store.getAll(keys) }
+            call.respondTextWriter(contentType = VersionDeltaStream.CONTENT_TYPE) {
```

Comment:
This will suffer from the same bugs in Ktor we worked around in https://github.com/modelix/modelix.core/pull/650
Consider using a similar mechanism to `VersionDeltaStreamV2`.

Reply:
I added an additional empty line to indicate the end of the stream. All these special characters as used in `VersionDeltaStreamV2` seems to be an unnecessary complex solution.

Reply:
Whitespaces are pretty easy to mess up and a trailing newline, while formally sufficient to resolve the problem, is not a very explicit marker and can easily be overlooked. I'd be more in favor of an explicit end marker as it is then less likely that the next one touching the code forgets about the reason for the trailing newline.

Reply:
I like that it's doesn't pollute the data with special characters and the response, when inspected, remains as easy to understand as before. The unit tests will fail if someone changes the code to not include the additional new line. Also, there are comments and error messages that document the reason.